asf_core_data/pipeline/data_joining/merge_install_dates.py

```diff
@@ -157,34 +114,38 @@ def manage_hp_install_dates(df, identifier="UPRN", verbose=False):
 
     df["FIRST_HP_MENTION"] = df[identifier].map(dict(first_hp_mention))
 
-    df["ANY_HP"] = df[identifier].map(
-        dict(df.groupby(identifier)["HP_INSTALLED"].max())
-    )
+    print(df[df["FIRST_HP_MENTION"].notna() & df["HP_INSTALLED"]].shape)
 
-    df["HP_AT_FIRST"] = df[identifier].map(
-        df.loc[df.groupby(identifier)["INSPECTION_DATE"].idxmin()]
-        .set_index(identifier)
-        .to_dict()["HP_INSTALLED"]
-    )
+    # Additional features about heat pump history of property
+    if add_hp_features:
 
-    df["HP_AT_LAST"] = df[identifier].map(
-        df.loc[df.groupby(identifier)["INSPECTION_DATE"].idxmax()]
-        .set_index(identifier)
-        .to_dict()["HP_INSTALLED"]
-    )
+        df["ANY_HP"] = df[identifier].map(
+            dict(df.groupby(identifier)["HP_INSTALLED"].max())
+        )
 
-    df["HP_LOST"] = df["HP_AT_FIRST"] & ~df["HP_AT_LAST"]
-    df["HP_ADDED"] = ~df["HP_AT_FIRST"] & df["HP_AT_LAST"]
-    df["HP_IN_THE_MIDDLE"] = (
-        ~df["HP_AT_FIRST"] & ~df["HP_AT_LAST"] & ~df["FIRST_HP_MENTION"].isna()
-    )
+        df["HP_AT_FIRST"] = df[identifier].map(
+            df.loc[df.groupby(identifier)["INSPECTION_DATE"].idxmin()]
+            .set_index(identifier)
+            .to_dict()["HP_INSTALLED"]
+        )
+
+        df["HP_AT_LAST"] = df[identifier].map(
+            df.loc[df.groupby(identifier)["INSPECTION_DATE"].idxmax()]
+            .set_index(identifier)
+            .to_dict()["HP_INSTALLED"]
+        )
```

Comment:
I find these a bit unreadable but I understand what you are doing.

Reply:
We don't really need them anymore, so probably not worth over-investing into this.